library/linkedList.js

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+class Node {
+  /**
+   * creating a node
+   * @param {any} data
+   * @return {node}
+   */
+  constructor(data) {
+    this.data = data;
+    this.next = null;
+  }
+
+  addLinkToBack(data) {
+    /**
+     * adding another link to the back
+     * eg. intial node => Node(data:1,next:null)
+     * new node => Node(data:1, next:(data:1, next:null))
+     * it runs a while loop till .next is null. Then inserts the new node such that .next: new node.
+     * @param {any} data
+     * @return {node}
+     */
+
+    let nextNode = this;
+    while (nextNode.next != null) {
+      nextNode = nextNode.next;
+    }
+    nextNode.next = new Node(data);
+    return this;
+  }
+
+  addLinkToFront(value) {
+    /**
+     * adding a node to the begging of the chain
+     * eg. intial node => Node(data:1, next:(data:2, next:null)) and value = "xyz"
+     * new node => Node(data:"xyz", next:(data:1, next:(data:2, next:null)))
+     * to achive the above, values of this.data and this.next is saved in the newNode => making newNode mimic this.
+     * then at this.data inserting value param and this.next's value to be the new node.
+     * @param {any} value
+     * @return {node}
+     */
+    let newNode = new Node();
+    newNode.data = this.data;
```

Comment:
can be shortened with `newNode = new Node(this.data)`

Reply:
Resolved.